Src/Sarif.PatternMatcher/AnalyzeCommand.cs

```diff
@@ -137,6 +137,16 @@ internal static SearchDefinitions PushInheritedData(SearchDefinitions definition
 
                 foreach (MatchExpression matchExpression in definition.MatchExpressions)
                 {
+                    if (matchExpression.SingleLineRegexes?.Count > 0)
+                    {
+                        for (int i = 0; i < matchExpression.SingleLineRegexes.Count; i++)
```

Comment:
nit: replace for:
for (int i = 0; i < matchExpression.SingleLineRegexes?.Count; i++)
{
     PushData
}

nit: verify the  alignment

Reply:
I fixed the alignment. I don't understand your other note. :) Don't forget, we already know because of line 140 that this member is non-null, which is why the '?' isn't used on line 142.

Reply:
similar to a change that u made in SearchSkimmer:
`for (int i = 0; i < matchExpression.IntrafileRegexes?.Count; i++)`

that would prevent the if + for.
that could be applied here as well.